nwbinspector/checks/tables.py

```diff
@@ -74,6 +75,34 @@ def check_time_intervals_stop_after_start(time_intervals: TimeIntervals, nelems:
         )
 
 
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=DynamicTable)
+def check_column_binary_capability(table: DynamicTable, nelems: int = 200):
+    """
+    Check each column of a table to see if the data could be set as a boolean dtype.
+
+    Parameters
+    ----------
+    time_intervals: TimeIntervals
+    nelems: int
+        Only check the first {nelems} elements. This is useful in case there columns are
+        very long so you don't need to load the entire array into memory. Use None to
+        load the entire arrays.
+    """
+    for column in table.columns:
+        if hasattr(column, "data") and not isinstance(column, VectorIndex):
+            unique_values = np.unique(column.data[:nelems])
+            saved_bytes = (unique_values.dtype.itemsize - 1) * np.product(
+                get_data_shape(data=column.data, strict_no_data_load=True)
+            )
+            if saved_bytes != 0 and len(unique_values) == 2:
```

Comment:
* How would you have 0 saved bytes with 2 unique values? Don't you then need at least 2 values?
* What about data this is of type bool?

Reply:
bools have an itemsize of 1, same as a `uint8`. Kind of an edge case there. Could still be best practice to reduce the `uint8` values to a `bool`, just because it makes it more obvious that there are only two possible values there.

Reply:
ok I see that this technically works correctly but for the sake of my sanity can you please change it to something like 

```python
if data.dtype.itemsize == 1:
    continue  # is already boolean, int8, or uint8
```